### PR TITLE
User can save a project as a draft

### PR DESCRIPTION
**Pull request by SARodrigues**
## Description

This PR adds the ability to save a new project as a draft, and preview its public page.  
The user can keep saving it as a Draft for as long as they need to, but once it's Published it cannot be turned into a draft again. The pending verification screen is only shown for published projects (or when publishing a new or drafted one). 

&nbsp; 

**In this PR**  

- `MultiPageLayout` changes: 
    - Added the ability to pass custom elements to the layout's footer
      _So we can add custom buttons, in this case the "Save as draft" one_  
    - Added a loading spinner screen
      _Which can be used while data to be used is still loading (eg: re-fetching, slow internet connections)_  
        - Footer buttons will be disabled while the spinner is being shown
    - Added an option so that we can show the website's static page header on any screen
      _Can be used in any screen, but the forms are set up to only display it in the Outro screen, to match the designs_  
- `ProjectForm` changes:  
    - Added a _Save as Draft_ button to the form footer, next to the _Submit_ button 
      _Only visible on the last page as long as the project hasn't been published yet_  
    - Extracted the logic related to where the user is redirected to after form completion to the respective new/edit pages  
      _To keep it consistent with the implementation done in the PD & Investor pages/forms done in #396_  
    - Reworked the "Outro" screen logic  
        - Will now only be shown when the user publishes a project, and the project is pending verification  
        - Will not be shown when the user is saving as draft, as it wouldn't be visible in the backoffice either    
        - Fixed the outro screens logic/redirection similar to the implementation on #396 
          _Although it's not really needed, due to the next point_
        - Fixed an issue where the "Pending verification" screen would display both the form and the static page header
          _The former hidden under the latter_
    - Fixed the "Pending verification" component's display
        - So it's centered in the page much like the "Pending approval" in the PD and Investor forms
- `Project` public page / preview  
    - Reworked the logic so that the page can also be used to preview unpublished projects  
      _Data loading logic changes are similar to the ones done for the PD and Investor public pages, in #342_
        - We still attempt to pre-render the project page, so that published projects work as before and are indexed by search engines
        - Unless previewing an unpublished project page, attempting to load an inexistent project page will still render a 404  
        - When previewing an unpublished project, the project will not be accessible from the endpoint so we load it client side (when the ownership can be verified and the endpoint returns the project)  
        - Project public pages can be previewed by either appending a `/preview` to the normal project public page url or a `?preview=true` query param. Logic is set to the former.  
        - Because when previewing a project public page, the data will be loaded in the client, a Loading spinner was added  
        - If a project is published, its preview page will not be accessible

**Other notes**  
- To keep all forms looking similar and better match the designs, the PD and Investor forms also show the site's static header when displaying the _Outro_ screen  
- Fixed some issues with updating a project twice consecutively loading the previous data
  _Not the best of solutions, but it was solved by invalidating the `Project` query completely when updating a project, so it'll be re-fetched_  
- Added ProjectList query invalidation when creating/updating a project
  _So that the dashboard's ProjectList is updated_  
- Removed the `ProjectQuery` enum, as it was a duplicate of the `Project` one  

## Testing instructions

**Saving as draft** 

1. Sign in as a Project Developer  
2. Visit the Dashboard and advance through the Project creation form but do not submit
    - Verify that validation works normally  
3. Click "Save as draft". Verify that:
    - The "Pending verification" screen does not show  
    - The user is redirected to dashboard/projects  
    - The newly created project is visible in the table  
    - The newly created project has a "Draft" status

**Previewing the project**  

1. From dashboard/projects, click on the three dots at the end of the project row  
2. Click on "Preview project page"  
    - Verify that the project page preview loads  

**Unpublished public page**  

1. Visit the project public page
  _Can be done easily by removing the `/preview`_ from the preview url  
    - Verify that the project public page doesn't load

**Publishing the project**  

1. Visit the dashboard/projects page  
2. Click on "Edit" next to the project  
3. Go through the form, this time click "Submit" to publish the project
    - Verify that the "Pending verification" screen is shown  
4. Click on "View project page" 
    - Verify that the project page loads  
5. Visit dashboard/projects and verify that:  
    - Verify that the project has a "Draft" status  
6. Click to edit and go through the form again:
    - Verify that at the last step, the "Save as draft" button is no longer  visible

## Tracking

[LET-775](https://vizzuality.atlassian.net/browse/LET-775)

### CODE REVIEW

**Comment by barbara-chaves** [comment]
> **Saving as draft**
> 
> 1. Sign in as a Project Developer
> 2. Visit the Dashboard and advance through the Project creation form but do not submit
>    
>    * Verify that validation works normally
> 3. Click "Save as draft". Verify that:
>    
>    * The "Pending verification" screen does not show
>    * The user is redirected to dashboard/projects
>    * The newly created project is visible in the table

  The project's table is not refreshed and the new project isn't visible

>    * The newly created project has a "Draft" status

When the page is refreshed the project can be seen and the status is correct

**Comment by barbara-chaves** [comment]
> **Previewing the project**
> 
> 1. From dashboard/projects, click on the three dots at the end of the project row
> 2. Click on "Preview project page"
>    
>    * Verify that the project page preview loads

The page shows a 404
it seems that using the preview as a path doesn’t  work ( /preview ), but when using a query (?preview=true) it works and load the page

Using the query:
Screenshot 2022-07-21 at 17 26 55

Using the path:
Screenshot 2022-07-21 at 17 27 04

**Comment by SARodrigues** [comment]
@barbara-chaves Thank you. 

Did you restart the `next` server and hard reloaded the page in the browser?  

The PR contains `next.config.js` changes that won't come into effect until the server is restarted.

**Comment by barbara-chaves** [comment]
> @barbara-chaves Thank you.
> 
> Did you restart the `next` server and hard reloaded the page in the browser?
> 
> The PR contains `next.config.js` changes that won't come into effect until the server is restarted.

You are right! It worked! I'm sorry

**Review by barbara-chaves — CHANGES_REQUESTED** [review]
Amazing work! The only thing to change is the getServerSideProps getEnum function placement, on  project/:id/edit page.

File: frontend/pages/project/[id]/edit.tsx [hunk 1]
```diff
@@ -23,29 +23,34 @@ import { User } from 'types/user';
...
-        'department',
-        'project_developer',
-        'involved_project_developers',
-        'project_developer',
-      ],
-      // We set the `locale` as `null` so that we get the project in the account's language instead of the UI language
-      locale: null,
-    }));
+    ({ data: project } = await getProject(id as string, PROJECT_QUERY_PARAMS));
     enums = await getEnums();
```
[the remark this answers is not in the document]
Comment (SARodrigues) [reply frontend/pages/project/[id]/edit.tsx:48]: Good catch, thank you Barbara! 

Added: https://github.com/Vizzuality/heco-invest/pull/404/commits/7388b4848a9f4aebfdd159447964eeff2b09f83b

**Review by barbara-chaves — APPROVED** [review]

**Pull request merged**